### PR TITLE
New 2.3 APIs

### PR DESCRIPTION
**Pull request by gjtorikian**
I [took a run through the new 2.3 APIs](https://developer.github.com/v3/enterprise/) and have implemented them in Octokit.rb. 

Closes https://github.com/octokit/octokit.rb/issues/636.

/cc @keavy @pengwynn

### CODE REVIEW

File: lib/octokit/enterprise_admin_client/users.rb [hunk 1]
```diff
@@ -48,6 +73,46 @@ def suspend(user, options = {})
...
+
+      # Creates an impersonation OAuth token.
+      #
+      # @param login [String] The user to create a token for.
+      # @param options [Array<String>] :scopes The scopes to apply.
+      # @see https://developer.github.com/v3/users/administration/#create-an-impersonation-oauth-token
+      # @example
+      #   @admin_client.create_impersonation_token('foobar', {:scopes => ['repo:write']})
+      def create_impersonation_token(login, options = {})
+        boolean_from_response :post, "admin/users/#{login}/authorizations", options
```
Comment (pengwynn) [inline lib/octokit/enterprise_admin_client/users.rb]: @gjtorikian I think we want the actual data here, no?

File: lib/octokit/enterprise_admin_client/users.rb [hunk 2]
```diff
@@ -5,6 +5,19 @@ class EnterpriseAdminClient
...
+      #
+      # @param login [String] The user's username.
+      # @param email [String] The user's email address.
+      # @see https://developer.github.com/v3/users/administration/#create-a-new-user
+      # @example
+      #   @admin_client.create_user('foobar', 'notreal@foo.bar')
+      def create_user(login, email, options = {})
+        options[:login] = login
+        options[:email] = email
+        boolean_from_response :post, "admin/users", options
```
Comment (pengwynn) [inline lib/octokit/enterprise_admin_client/users.rb]: Probably want the user hash here.

File: lib/octokit/enterprise_admin_client/users.rb [hunk 3]
```diff
@@ -27,6 +40,18 @@ def demote(user, options = {})
...
+      # Rename a user.
+      #
+      # @param old_login [String] The user's old username.
+      # @param new_login [String] The user's new username.
+      # @see https://developer.github.com/v3/users/administration/#rename-an-existing-user
+      # @example
+      #   @admin_client.rename_user('foobar', 'foofoobar')
+      def rename_user(old_login, new_login, options = {})
+        options[:login] = new_login
+        boolean_from_response :patch, "admin/users/#{old_login}", options
```
Comment (pengwynn) [inline lib/octokit/enterprise_admin_client/users.rb]: And here?

**Comment by gjtorikian** [comment]
Ugh sorry about those @pengwynn. Probably some copy-pasta blindness. :pray:

**Comment by pengwynn** [comment]
Thanks @gjtorikian!

**Pull request merged**